### DISCUSSION TITLE
X-Data for Laravel Eloquent Relationship

### DISCUSSION THREAD

**Discussion by seanlabs** [alpinejs/alpine#763] [4. General]
I am trying to figure out how to assign an x-data element to an eloquent relationship. 

I have 2 models (Post and Comments). A Post hasMany comments and a Comment belongsTo a Post. I am trying to pull in the Post model and associated Comments into a Alpine JS x-data. I am able to pull in the Posts and increment over them with x-for, but I cannot figure out how to create the data structure for the related comments. 

I tried doing a @foreach with posts and creating a comments array, but this doesn't seem to work.

Any ideas?

```
  <section
        x-data="{
            posts: {{$posts}},
	    @foreach($posts as $post)
                comments[$post->id]: {{$post->comments}}
            @endforeach		
}">
```

**Comment by SimoTod** [comment]
It's a Laravel question but if you eager load those comments, they'll be serialised with your posts. At that point, you just access them using a x-for in alpine.

**Reply by seanlabs** [reply]
Thank you very much.
